docs_nnx/guides/filters_guide.ipynb

```diff
@@ -5,12 +5,18 @@
    "id": "95b08e64",
    "metadata": {},
    "source": [
-    "# Using Filters\n",
+    "# Using `Filter`s\n",
```

Comment:
I can't help but think there could be better names than this, because simply saying "filter" doesn't give much info.
What about "Grouping NNX variables"?

Also a note: we are about to remove `nnx.State` in https://github.com/google/flax/pull/4418. This means there could be another overhaul of this doc that removes the `nnx.State` mentions.

Reply:
Good idea. For now, let's keep "Using Filters" and add your suggestion "Grouping NNX variables":

```
# Using Filters, grouping NNX variables
```

Can update this later.

@cgarciae @IvyZX